### ISSUE TITLE
Unable to reclaim orphan calendar entries in Google calendar.

### ISSUE THREAD

**Issue by mrosing** [phw198/OutlookGoogleCalendarSync#422]
:memo: Use the below template to report your issue.  
:bulb: Have you upgraded to the latest alpha to check if the problem is already fixed?  
:paperclip: To aid investigation, please provide your OGcalsync.log file ([see instructions](https://github.com/phw198/OutlookGoogleCalendarSync/wiki/Reporting-problems#posting-your-ogcalsynclog-file)).  

----

**OGCS Version**: v2.6.0  
**Installed or Portable**: Installed

**Problem Description**  
Repeatedly get "Unable to reclaim orphan calendar entries in Google calendar."...Your server administrator has limited the number of items you can open simultaneously.

**Steps to Reproduce the Issue**  
1. Every sync

**Any other information**  
Disconnected Google calendar and reconnected, rebooted PC, at a loss what else to do.

**Comment by phw198** [comment]
What version of OGCS are you running? 
Was it working OK before, or have you only just started using OGCS?
What date range are you trying to sync? Reduce it?

**Comment by mrosing** [comment]
2.6.0.0
It was working before and now has stopped.
Thanks!
Mark

**Comment by phw198** [comment]
Try upgrading to the latest alpha release then?

**Comment by mrosing** [comment]
How do I find the download, please?



Mark A. Rosing, MD, MPH
Chair of Obstetrics & Gynecology, SBH Health System, Bronx, NY


-------- Original message --------

**Comment by phw198** [comment]
OGCS > About > Check "Check for Alpha releases" > Check for Update

**Comment by mrosing** [comment]
Paul,
It says that I’m using the latest version (2.6.0.0)

Mark A. Rosing, MD, MPH

**Comment by phw198** [comment]
Ah yes, a peculiarity around the migration to GitHub that will get resolve come the next Beta release (which shouldn't be long). In the meantime, [download the latest alpha setup.exe](https://github.com/phw198/OutlookGoogleCalendarSync/releases/download/v2.6.5-alpha/Setup.exe).

**Comment by mrosing** [comment]
I tried it.  Same… ??
[cid:image001.png@01D3887C.F797F690]

Mark A. Rosing, MD, MPH

**Comment by phw198** [comment]
:paperclips: Unfortunately, files and images can't be attached to the ticket when replying by email - please log in to GitHub to upload them.

**Comment by mrosing** [comment]
[OGcalsync.log](https://github.com/phw198/OutlookGoogleCalendarSync/files/1612527/OGcalsync.log)

I hope this is what you need.
[OGcalsync.log](https://github.com/phw198/OutlookGoogleCalendarSync/files/1612549/OGcalsync.log)
Thanks for your efforts!

**Comment by phw198** [comment]
You have your Exchange connection set to "Online" - did it used to be "Cached"? Either way, can you try setting it to this?

Outlook > Account Settings > Change... > Check "Use Cached Exchange Mode".

**Comment by mrosing** [comment]
I'm running this on my hospital PC, the only PC where I have outlook running in MS outlook on the hospital network.  Other places (i.e. tablet and android) I gain access only through web owa.
I'm a bit unsure what the ramifications of running in Cached mode would be on functionality at this primary site?  I do have admin rights so I can do it.  Please advise before I do so (It worked before and it was not run in Cached mode).

**Comment by phw198** [comment]
Cached mode is the [default and preferred option](https://support.office.com/en-gb/article/turn-on-cached-exchange-mode-7885af08-9a60-4ec3-850a-e221c1ed0c1c), according to Microsoft - so if you have the option, I'm not sure why you would _not_ want to run cached. Cached mode means you have local copy of your mailbox and aren't prone to network drop outs. It also means you don't need an active connection to the Exchange server for pretty much every action you take within Outlook - and this is the bit that's key for OGCS.

It's possible for an Exchange admin to disable cached mode, though. For a slightly more technical explanation, [see here](https://technet.microsoft.com/en-us/library/jj683103.aspx). It may be possible that given you work in the health sector, they don't want a cached copy of your emails to reside on your computer hard drive, and only be on the central server. If that's the case and you can't enable cached mode, then you're only option would be to reduce the sync date range....or pester your Exchange admins to increase the number of items you can access at the same time.

**Comment by mrosing** [comment]
I never got back an answer about the impact of cached mode (I do have admin rights) but I did change my sync range to 1mo from 45d and I also had an issue with some repeating events (i.e. birthdays) that were even outside of the 30d range (years out!) and so I removed earlier iterations of those events. (I AM NOT SURE WHY THIS WOULD BE) and it worked!
I'd love to understand why 

- 30d worked and 45 didn't?
Why repeating event would cause a problem?
If there is any issue with running sync more frequently in terms of creating duplicates (I have somehow generated some duplicates even though I'm set to only sync outlook -> google and checked off merge with existing entries?

Thanks so much!

 worked and 45 didn't

**Comment by phw198** [comment]
You didn't get an answer back from who about cached mode? You certainly did from me.

Why are you surprised about recurring events? A recurring event consists of a _master_ appointments (the first in the series) and that it's - the rest are pseudo appointments. If you want those pseudo appointments to be synced if they appear in your synced date range, you have to sync the original, which yes, could be 100 years ago.

Reducing the sync range from 45 to 30 days would be the expected fix if you don't enable cached mode.

If you've ended up with duplicates and are only syncing one-way to a dedicated sync calendar, then turn off merge, enable deletes and press and hold `Shift` while clicking the `Sync` button.

**Comment by guitaradam2000** [comment]
Been having the same problem. Running version 2.7.8.0
My hotmail outlook account is in cached mode, Gmail account doesn't give me that option.
Tried changing sync times to 30 days, but still no joy. I have a 2-way sync set-up, seems to be failing when trying to perform the 'Google to Outlook' sync:

<img width="393" alt="OGCS sync error" src="https://user-images.githubusercontent.com/47711281/56093464-7e04c980-5f0c-11e9-8f8e-dd34740e89e0.PNG">

**Comment by phw198** [comment]
@guitaradam2000 Your issue should be fixed over on #689 - [try the hotfix there](https://github.com/phw198/OutlookGoogleCalendarSync/issues/689#issuecomment-480601660) and post your findings on the other ticket please.

**Comment by guitaradam2000** [comment]
Tried the hotfix version, I followed the instructions but the app doesn't run at all. Nothing happens. If I rename the original from OutlookGoogleCalendarSync.exe.old back to OutlookGoogleCalendarSync.exe, the application starts running again. 
Tried on 2 computers, same result. I am running it from: C:\Users\%username%\AppData\Local\OutlookGoogleCalendarSync

**Comment by phw198** [comment]
@guitaradam2000 Which file exactly are you replacing with the hotfix?

`%LOCALAPPDATA%\OutlookGoogleCalendarSync\OutlookGoogleCalendarSync.exe`
or 
`%LOCALAPPDATA%\OutlookGoogleCalendarSync\app-v2.7.8-alpha\OutlookGoogleCalendarSync.exe`

The instructions say the second location (and I've just edited it to make that clearer)...

**Comment by guitaradam2000** [comment]
@phw198 You're right, I had replaced the file in the wrong location. After replacing it in the "app-2.7.8-alpha" the application does run now.

I also changed the sync down to 1 day and the original errors have cleared. Just left with 1 error now:
"Appointment update failed. The recurrence pattern is not valid."

I'm sure this is a problem with a single appointment. Any way I can find out which is the trouble-maker?

Thanks
